### ISSUE TITLE
The library appears to reorder elements?

### ISSUE THREAD

**Issue by mitchellwrosen** [nikita-volkov/acc#10]
Hi, apologies if I'm not understanding how to use the API properly, but in playing around, I believe I've encountered a bug.

```haskell
> Acc.unsnoc $ Acc.cons 2 $ Acc.cons 1 $ Acc.cons 1 mempty
Just (1,[1,2])
```

I believe the correct answer here is

```haskell
Just (1,[2,1])
```

i.e. the original `Acc` minus the rightmost element. Thanks.

**Comment by nikita-volkov** [comment]
Thanks for the report! I'll check.

**Comment by nikita-volkov** [comment]
Thanks. It was indeed a bug. It's been fixed and released as 0.2.0.3
